felix/dataplane/linux/int_dataplane.go

```diff
@@ -765,6 +766,26 @@ func NewIntDataplaneDriver(config Config) *InternalDataplane {
 		dp.iptablesMangleTables = append(dp.iptablesMangleTables, mangleTableV6)
 		dp.iptablesFilterTables = append(dp.iptablesFilterTables, filterTableV6)
 
+		if config.RulesConfig.VXLANEnabled {
+			routeTableVXLAN := routetable.New([]string{"^vxlan-v6.calico$"}, 6, true, config.NetlinkTimeout,
+				config.DeviceRouteSourceAddressIPv6, config.DeviceRouteProtocol, true, unix.RT_TABLE_UNSPEC,
+				dp.loopSummarizer, routetable.WithLivenessCB(dp.reportHealth))
+
+			vxlanManager := newVXLANManager(
+				ipSetsV6,
+				routeTableVXLAN,
+				"vxlan-v6.calico",
+				config,
+				dp.loopSummarizer,
+				6,
+			)
+			go vxlanManager.KeepVXLANDeviceInSync(config.VXLANMTU, iptablesFeatures.ChecksumOffloadBroken, 10*time.Second)
+			dp.RegisterManager(vxlanManager)
+		} else {
+			// Start a cleanup goroutine not to block felix if it needs to retry
+			go cleanUpVXLANDevice()
```

Comment:
(Apologies for looking at this in multiple passes) I think we need to have `deviceName` as an arg to `cleanUpVXLANDevice()` (specifically for the call to `netlink.LinkByName()` in it), and to pass the v4/v6 device names on their respective `else` cases...

Reply:
Good catch, I hadn't delved too far into it and assumed that it did a prefix match for `vxlan`. I'll fix that up.